cpp/src/dataframe/cudf_table.cuh

```diff
@@ -467,7 +467,7 @@ public:
     {
```

Comment:
The caller does not know the type of either type-erased `void*` column data buffers. This is why the type_dispatcher is used, to dispatch the type to the `copy_element` functor such that the type-erased buffers can be casted to the appropriate type.

Reply:
But the _direct_ caller of this templated operator() is the type dispatcher... which does know. So this operator can take typed pointers.

Reply:
All the `type_dispatcher` knows about is an parameter pack that can be any number or any type of arguments. How will it know which arguments are what?

Reply:
Hmm. Yeah, I forgot about that part.

But - we _can_ get around this restriction: Instead of simple perfect-forwarding, we could static-cast cast the void pointers in the parameter pack to the types in the signature, which we can extract. This should not be terribly difficult to code - instead of std::forward` we'll need to transform the tuple using `F` and an index sequence. Do you follow what I'm suggesting or is it too vague?

Reply:
I'm fuzzy on the implementation details, but I understand that you're suggesting we essentially iterate the parameter pack, find all the `void*` arguments, cast them to `T*`, and pass them on as typed pointers to the functor.

One thing I'm worried about this approach is that it would blindly cast any `void*` to a typed pointer. What if a user of `type_dispatcher` passed in a `void*` argument they _don't_ want casted? (I don't know why this would be the case, but I can't predict the future use cases.) 

I think it's safer to just keep the `void*` as is, and let the user of `type_dispatcher` get the dispatched type and cast whatever pointers they need to be casted.

Reply:
How can I use call `foo(T*)` with a `void *` and not want it to be cast?

Reply:
I should have specified that the user may not want the `void*` casted to a `T*`. Inside their functor, they may want to cast it to some other type, `Y*`. 

Point is, we don't know, and I don't think we should design a utility that forces this behavior. 

Let the programmer be responsible for doing the appropriate casting.